pkg/clusteragent/admission/controllers/webhook/controller_base.go

```diff
@@ -118,38 +111,96 @@ func (c *controllerBase) generateWebhooks(wmeta workloadmeta.Component, pa workl
-		}
-		webhooks = append(webhooks, mutatingWebhooks...)
+	// Skip mutating webhooks if the mutation feature is disabled.
+	if !c.config.isMutationEnabled() {
+		return webhooks
+	}
+
+	// Setup config webhook.
+	configWebhook, err := generateConfigWebhook(wmeta, datadogConfig)
+	if err != nil {
+		log.Errorf("failed to register config webhook: %v", err)
+	} else {
+		webhooks = append(webhooks, configWebhook)
+	}
+
+	// Setup tags from labels webhook.
+	tagsWebhook, err := generateTagsFromLabelsWebhook(wmeta, datadogConfig)
+	if err != nil {
+		log.Errorf("failed to register tags from labels webhook: %v", err)
+	} else {
+		webhooks = append(webhooks, tagsWebhook)
+	}
+
+	// Setup agents sidecar webhook.
+	agentsWebhook := agentsidecar.NewWebhook(datadogConfig)
+	webhooks = append(webhooks, agentsWebhook)
+
+	// Setup autoscaling webhook.
+	autoscalingWebhook := autoscaling.NewWebhook(pa, datadogConfig)
+	webhooks = append(webhooks, autoscalingWebhook)
+
+	// Setup APM Instrumentation webhook. APM Instrumentation webhook needs to be registered after the config webhook.
+	apmWebhook, err := generateAutoInstrumentationWebhook(wmeta, datadogConfig)
+	if err != nil {
+		log.Errorf("failed to register APM Instrumentation webhook: %v", err)
+	} else {
+		webhooks = append(webhooks, apmWebhook)
+	}
 
-		// APM Instrumentation webhook needs to be registered after the configWebhook webhook.
-		apm, err := autoinstrumentation.NewWebhook(wmeta, datadogConfig, injectionFilter)
+	isCWSInstrumentationEnabled := datadogConfig.GetBool("admission_controller.cws_instrumentation.enabled")
+	if isCWSInstrumentationEnabled {
+		cws, err := cwsinstrumentation.NewCWSInstrumentation(wmeta, datadogConfig)
 		if err == nil {
-			webhooks = append(webhooks, apm)
+			webhooks = append(webhooks, cws.WebhookForPods(), cws.WebhookForCommands())
 		} else {
-			log.Errorf("failed to register APM Instrumentation webhook: %v", err)
-		}
-
-		isCWSInstrumentationEnabled := datadogConfig.GetBool("admission_controller.cws_instrumentation.enabled")
-		if isCWSInstrumentationEnabled {
-			cws, err := cwsinstrumentation.NewCWSInstrumentation(wmeta, datadogConfig)
-			if err == nil {
-				webhooks = append(webhooks, cws.WebhookForPods(), cws.WebhookForCommands())
-			} else {
-				log.Errorf("failed to register CWS Instrumentation webhook: %v", err)
-			}
+			log.Errorf("failed to register CWS Instrumentation webhook: %v", err)
 		}
 	}
 
 	return webhooks
 }
 
+func generateConfigWebhook(wmeta workloadmeta.Component, datadogConfig config.Component) (*configWebhook.Webhook, error) {
+	filter, err := configWebhook.NewFilter(configWebhook.NewFilterConfig(datadogConfig))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create config filter: %v", err)
+	}
+	mutatorCfg := configWebhook.NewMutatorConfig(datadogConfig)
+	mutator := configWebhook.NewMutator(mutatorCfg, filter)
+	return configWebhook.NewWebhook(wmeta, datadogConfig, mutator), nil
```

Comment:
This might be out of scope for this change, but do the `NewWebhok` functions need to have the `datadogConfig` passed in anymore?